modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_6_1/DDMFieldAttributeUpgradeProcess.java

```diff
@@ -185,6 +189,14 @@ private long _getDLFileEntryId(long companyId, String src)
 			try {
 				DLFileEntry dlFileEntry = _getDLFileEntry(companyId, matcher);
 
+				if (dlFileEntry == null) {
```

Comment:
@adolfopa, resolveFriendlyURL in _getDLFileEntry uses 

`_dlAppLocalService.getFileEntry(friendlyURLEntry.getClassPK()); 
`
which throws an exception if not found. So what we are checking would be if the friendlyURLEntry exists right?

Reply:
The friendly URL entry may be null if the document doesn't exist:
```
	@Override
	public FileEntry resolveFriendlyURL(long groupId, String friendlyURL)
		throws PortalException {

		FriendlyURLEntry friendlyURLEntry =
			_friendlyURLEntryLocalService.fetchFriendlyURLEntry(
				groupId, FileEntry.class, friendlyURL);

		if (friendlyURLEntry == null) {
			return null;
		}

		return _dlAppLocalService.getFileEntry(friendlyURLEntry.getClassPK());
	}
```
This may happen is you delete a document and there are existing WC that contain URLs pointing to it.